### ISSUE TITLE
atomic-ops.h unconditionally defines instruction for ppc64 which is invalid on Darwin and causes build to fail

### ISSUE THREAD

**Issue by barracuda156** [libuv/libuv#3623]
I am trying to build `libuv` @1.44.1 as FAT (ppc+ppc64), and it consistently fails on the following:
```
{standard input}:28:Parameter syntax error (parameter 1)
make: *** [src/unix/libuv_la-async.lo] Error 1
make: *** Waiting for unfinished jobs....
```
Tried with `gcc-4.2`, `gcc10` @10.3.0 and `gcc11` @11.3.0. The same error persists.

More details here: https://trac.macports.org/ticket/64262 (no solution).
macOS 10.5.8, Xcode 3.1.4

**Comment by bnoordhuis** [comment]
Is there reason to believe that's an issue with libuv itself and not e.g. the version of autotools or libtool you're using?

**Comment by vtjnash** [comment]
I believe we deprecated support for that platform many years ago

**Comment by barracuda156** [comment]
UPD. I have pulled the master repo and tried building it with Cmake. Unfortunately, it still fails with:
```
{standard input}:28:Parameter syntax error (parameter 1)
make[2]: *** [CMakeFiles/uv_a.dir/src/unix/async.c.o] Error 1
{standard input}:28:Parameter syntax error (parameter 1)
make[2]: *** [CMakeFiles/uv.dir/src/unix/async.c.o] Error 1
```
`cmake` @3.22.2 + `gcc` @11.3.0

**Comment by barracuda156** [comment]
Well, I have found the problem – invalid `ppc64` instruction in `atomic-ops.h`:
```
#elif defined(__powerpc64__) || defined(__ppc64__) || defined(__PPC64__)
  __asm__ __volatile__ ("or 1,1,1; or 2,2,2" ::: "memory");
#endif
```
Let me see if I can make a version that gonna work for Darwin ABI (should be trivial, I just don't know assembler, so need to look), but as a quick fix, this works:
```
#elif !defined(__APPLE_) && (defined(__powerpc64__) || defined(__ppc64__) || defined(__PPC64__))
  __asm__ __volatile__ ("or 1,1,1; or 2,2,2" ::: "memory");
#endif
```
Now:
```
The following ports are currently installed:
  libuv @1.42.0_1 requested_variants='-universal' platform='darwin 9' archs='ppc' date='2021-12-22T08:43:53+0800'
  libuv @1.43.0_0 requested_variants='-universal' platform='darwin 9' archs='ppc' date='2022-01-09T04:36:52+0800'
  libuv @1.44.1_0 requested_variants='-universal' platform='darwin 9' archs='ppc' date='2022-04-08T22:25:08+0800'
  libuv @1.44.1_0+universal (active) requested_variants='+universal' platform='darwin 9' archs='ppc ppc64' date='2022-05-12T18:55:26+0800'
```

@vtjnash @bnoordhuis Do you mind if I submit a PR for this? I want to find correct atomics for Darwin ABI, so please allow me a couple of days. But in the worst case at least making sure ELF syntax is not used on Darwin is good enough :)

**Comment by bnoordhuis** [comment]
@barracuda156 Sure, go ahead. You can turn cpu_relax() into a no-op if you don't find the right incantation.

**Comment by barracuda156** [comment]
> @barracuda156 Sure, go ahead. You can turn cpu_relax() into a no-op if you don't find the right incantation.

@bnoordhuis I have opened a PR now: https://github.com/libuv/libuv/pull/3634
(Sorry for a delay, been busy with other project.)

**Comment by bnoordhuis** [comment]
This was fixed in 99ab53e but, mea culpa, I forgot to add a Fixes: tag.
